internal/service.go

```diff
@@ -409,47 +253,127 @@ func (h *HauserService) BackoffOnError(err error) bool {
 	return false
 }
 
-func (h *HauserService) Init(_ context.Context) error {
+func (h *HauserService) Init(ctx context.Context) error {
 	// Initialize the warehouse's schema
 	if !h.config.StorageOnly {
-		if err := h.database.EnsureCompatibleExportTable(); err != nil {
+		return h.InitDatabase(ctx)
+	}
+	return nil
+}
+
+func (h *HauserService) InitDatabase(_ context.Context) error {
+	if created, err := h.database.InitExportTable(h.schema); err != nil {
+		return err
+	} else if !created {
+		existingCols := h.database.GetExportTableColumns()
+		newSchema := h.schema.ReconcileWithExisting(existingCols)
+		if err := h.database.ApplyExportSchema(newSchema); err != nil {
 			return err
 		}
-		// NB: We SHOULD fetch the table columns ONLY after the call to EnsureCompatibleExportTable.
-		// The EnsureCompatibleExportTable function potentially alters the schema of the export table in the client warehouse.
-		h.tableColumns = h.database.GetExportTableColumns()
-		for i, col := range h.tableColumns {
-			h.tableColumns[i] = strings.ToLower(col)
-		}
-	} else {
-		h.tableColumns = warehouse.DefaultBundleColumns()
+		h.schema = newSchema
 	}
 	return nil
 }
 
-func (h *HauserService) ProcessNext(ctx context.Context) (int, error) {
+// ProcessNext will return the number of
+func (h *HauserService) ProcessNext(ctx context.Context) (time.Duration, error) {
 	lastSyncedRecord, err := h.lastSyncPoint(ctx)
 	if err != nil {
 		return 0, err
 	}
-	return h.ProcessExportsSince(ctx, lastSyncedRecord)
+
+	if lastSyncedRecord.IsZero() {
+		// If starting fresh, use the start time provided in the config
+		lastSyncedRecord = h.config.StartTime
+	}
+
+	// We need to ensure that the end time for the export is aligned with the export duration
+	nextEndTime := lastSyncedRecord.Add(h.config.ExportDuration.Duration).Truncate(h.config.ExportDuration.Duration)
+	for {
+		lastAvailableEndTime := getNow().Add(-1 * h.config.ExportDelay.Duration)
+		if nextEndTime.After(lastAvailableEndTime) {
+			waitUntil := nextEndTime.Add(h.config.ExportDelay.Duration)
+			return waitUntil.Sub(getNow()), nil
+		} else {
+			break
+		}
+	}
+
+	log.Printf("Creating export for %s to %s", lastSyncedRecord, nextEndTime)
+	id, err := h.fsClient.CreateExport(lastSyncedRecord, nextEndTime, h.schema.GetFullStoryFields())
+	if err != nil {
+		return 0, err
+	}
+
+	exportId := ""
+	var prog int
+	for {
+		prog, exportId, err = h.fsClient.GetExportProgress(id)
+		if err != nil {
+			return 0, err
+		}
+		log.Printf("Export progress: %d%%", prog)
+		if exportId != "" {
+			break
+		}
+		time.Sleep(progressPollDuration)
+	}
+
+	log.Printf("Fetching export for operation id %s", id)
+	body, err := h.fsClient.GetExport(exportId)
+	if err != nil {
+		return 0, err
+	}
+	defer body.Close()
+
+	unzipped, err := gzip.NewReader(body)
```

Comment:
No, but I didn't expect it to based on the headers that we set. Did you expect automatic decompression to happen?

Reply:
Yeah, I would have thought it did from the default http transport. See docs for `DisableCompression` option on transport (we don't disable it though, so I would have thought we would have got the automatic behavior?).

Reply:
So, the default client does set the `Accept-Encoding` header to `gzip`, but we only supply the `Content-Encoding` header for the "legacy user list" export since we can guarantee it's only one file. Otherwise, we set the `Content-Type` to `application/gzip` which shouldn't auto-unzip.

Reply:
Ohhhh yeah.. that's right. Okay that makes more sense why this wasn't automatic now.